### PR TITLE
Added the source for the 'Autumn Leaves' particle system

### PR DESCRIPTION
**Pull request by J-128**
Added:
        /src/object/leaves_particle_system.cpp
        /src/object/leaves_particle_system.hpp

### CODE REVIEW

File: src/object/leaves_particle_system.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  parse(reader);
+}
+
+LeavesParticleSystem::~LeavesParticleSystem()
+{
+}
+
+void LeavesParticleSystem::init()
+{
+  leavesimages[0] = Surface::create("images/objects/particles/leaf5.png");
```
Comment (tobbi) [inline src/object/leaves_particle_system.cpp]: for(int i = 0; i <= 5; i++)
leavesimages[i] = Surface::create()

**Comment by maxteufel** [comment]
This is basically using the same code as snow particles do, so please deduplicate the code.

**Comment by J-128** [comment]
@maxteufel Well, of course it uses code similar to the snow particles, because I copied the snow particle code, and adapted it to the leaves. But I don't know about de-duplicating it. I don't know how I'd do that. :confused:

**Comment by maxteufel** [comment]
You inherit from a common class. This is a minimum amount of effort for such code changes, I really don't want to accept this without that change. Would be really bad code style otherwise.

**Comment by Karkus476** [comment]
@J-128 Create a class for all particle systems and allow the programmer to create sub classes which define what images are displayed and other variables if they are possible. Then create two of these, one for snow and one for leaves.

**Comment by J-128** [comment]
Closing this for now, not making any headway on this.

**Pull request closed without merging**